cypress/tests/integration/Doi.spec.js

```diff
@@ -119,110 +68,83 @@ describe('DOI tests', function() {
-		cy.get('.app__notifications').contains('Items successfully marked registered', {timeout:20000});
+		cy.checkDoiFilterResults('Submitted', 'No items found.', 0);
+		cy.checkDoiFilterResults('Registered', 'Karbasizaed — Antimicrobial, heavy metal resistance and plasmid profile of coliforms isolated from nosocomial infections in a hospital in Isfahan, Iran', 1);
+	});
 
-		cy.get(`#list-item-submission-${submissionId} .pkpBadge`).contains('Registered');
+	it('Check unpublished Issue Marked Registered displays error', function() {
+		loginAndGoToDoiPage('issue');
+		cy.checkDoiMarkedStatus('Registered', unpublishedIssueId, false, 'Unpublished', 'issue');
 	});
 
-	it('Check Issue Filter Behaviour (post-deposit)', function() {
-		cy.login('dbarnes', null, 'publicknowledge');
+	it('Check unpublished Submission Marked Registered displays error', function() {
+		loginAndGoToDoiPage();
 
-		cy.get('a:contains("DOIs")').click();
-		cy.get('button#issue-doi-management-button').click();
+		cy.checkDoiMarkedStatus('Registered', unpublishedSubmissionId, false, 'Unpublished');
+	});
 
-		// Submitted
-		cy.get('#issue-doi-management button:contains("Submitted")').click();
-		cy.get('#issue-doi-management .listPanel__items').contains('No items found.');
+	it('Check Issue Marked Stale', function() {
+		loginAndGoToDoiPage('issue')
+		cy.checkDoiMarkedStatus('Stale', issueId, true, 'Stale', 'issue');
+	});
 
-		// Registered
-		cy.get('#issue-doi-management button:contains("Registered")').click();
-		cy.contains('Vol. 1 No. 2 (2014)');
+	it('Check Submission Marked Stale', function() {
+		loginAndGoToDoiPage();
+		cy.checkDoiMarkedStatus('Stale', submissionId, true, 'Stale');
 	});
 
-	it('Check Submission Filter Behaviour (post-deposit)', function() {
-		cy.login('dbarnes', null, 'publicknowledge');
+	it('Check Issue Marked Unregistered', function() {
+		loginAndGoToDoiPage('issue');
+		cy.checkDoiMarkedStatus('Unregistered', issueId, true, 'Unregistered', 'issue');
+	});
 
-		cy.get('a:contains("DOIs")').click();
-		cy.get('button#article-doi-management-button').click();
+	it('Check Submission Marked Unregistered', function() {
+		loginAndGoToDoiPage();
+		cy.checkDoiMarkedStatus('Unregistered', submissionId, true, 'Unregistered');
+	});
 
-		// Submitted
-		cy.get('#article-doi-management button:contains("Submitted")').click();
-		cy.get('#article-doi-management .listPanel__items').contains('No items found.');
+	it('Check invalid Issue Marked Stale displays error', function() {
+		loginAndGoToDoiPage('issue');
+		cy.checkDoiMarkedStatus('Stale', issueId, false, 'Unregistered', 'issue');
+	});
 
-		// Unregistered
-		cy.get('#article-doi-management button:contains("Registered")').click();
-		cy.contains(' Karbasizaed — Antimicrobial, heavy metal resistance and plasmid profile of coliforms isolated from nosocomial infections in a hospital in Isfahan, Iran ');
+	it('Check invalid Submission Marked Stale displays error', function() {
+		loginAndGoToDoiPage();
```

Comment:
Having all of the tests broken down into small things is good practice generally. However, for us, logging in and browsing to the DOI page will probably cost 5-10 seconds per call on Travis. That means the tests might be about 2 minutes longer just from the repeated `loginAndGoToDoiPage()` calls.

That doesn't seem like much, but it does add up. Currently our tests are coming in at about 30-35 minutes, so it's about 5% of the overall build time.

In this case, I think I'd prefer to reduce the number of logins and run multiple checks for each `it()` function.